### DISCUSSION TITLE
Failing to see how to persist profile info in JWT based session

### DISCUSSION THREAD

**Discussion by pshort** [nextauthjs/next-auth#10462] [Help]
Hi all,

I am doing OIDC with a local duende server in auth.js beta. I am using the callbacks to get the profile info from the token at the end of the flow but I am having a hard time getting typescript to let me persist this info into the session.

I see 2 callbacks jwt({token, profile, trigger}) and session({session, token}) that I can use and what I am doing is the following:

```typescript
    async jwt({ token, account, session, trigger, profile }) {
      if (trigger === "signIn") {
        token.profile = profile;
      }
      return token;
    },
    async session({ session, token }) {
      session.user = token.profile;
      return session;
    },
```

Typescript really doesnt want me to do this - it thinks session is some other shape:

    
``` bash
  Type error: Type 'unknown' is not assignable to type 'AdapterUser & User'.
  Type 'unknown' is not assignable to type 'AdapterUser'.

  50 |     },
  51 |     async session({ session, token }) {
> 52 |       session.user = token.profile;
     |       ^
```
    

I have the session strategy as "jwt" so Im not sure how to get ts to choose the other params type, as this looks the database one. Can anyone give me a pointer in the right direction?

Thank you!

**Comment by ntdoJanneck** [comment]
Hey, I'm currently having a similar issue when augmenting the JWT and session types to include custom properties (https://github.com/nextauthjs/next-auth/discussions/12674) with the Discord provider. Did you manage to solve this?

Edit: nevermind, I just solved my issue myself. See my linked discussion 😅
